lib/site_settings/hidden_provider.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module SiteSettings
+end
+
+# A cache for providing default value based on site locale
+class SiteSettings::HiddenProvider
+  def initialize
+    @hidden_settings = []
```

Comment:
```suggestion
    @hidden_settings = Set.new
```

I think using a set here will be better since we need things to be unique.